pkgs/tools/misc/dvtm/default.nix

```diff
@@ -1,20 +1,28 @@
-{ stdenv, fetchurl, ncurses }:
+{ stdenv, fetchurl, ncurses, customConfig ? null }:
 
 stdenv.mkDerivation rec {
+
   name = "dvtm-0.15";
 
-  meta = {
+  meta = with stdenv.lib; {
     description = "Dynamic virtual terminal manager";
     homepage = http://www.brain-dump.org/projects/dvtm;
-    license = stdenv.lib.licenses.mit;
-    platfroms = stdenv.lib.platforms.linux;
+    license = licenses.mit;
+    platfroms = platforms.linux;
+    maintainers = [ maintainers.vrthra ];
   };
 
   src = fetchurl {
     url = "${meta.homepage}/${name}.tar.gz";
     sha256 = "0475w514b7i3gxk6khy8pfj2gx9l7lv2pwacmq92zn1abv01a84g";
   };
 
+  postPatch = stdenv.lib.optionalString (customConfig != null) ''
+    cat <<EOF > config.h
+    ${customConfig}
```

Comment:
This seems fragile to me.
Maybe consider `cp ${builtins.toFile "config.h" customConfig} ./config.h`?